src/webapp/components/theme-list/ThemeListTable.tsx

```diff
@@ -32,20 +37,48 @@ export interface ThemeDetail {
     subtitle: string;
     logo: ReactNode;
     palette: string[];
+    users: SharingRule[];
+    userGroups: SharingRule[];
 }
 
 type ThemeListTableProps = Pick<RouteComponentProps, "themes" | "setThemes">;
 
 export default function ThemeListTable({ themes, setThemes }: ThemeListTableProps) {
-    const { compositionRoot } = useAppContext();
+    const { api, compositionRoot } = useAppContext();
     const snackbar = useSnackbar();
     const loading = useLoading();
 
     const [selection, setSelection] = useState<TableSelection[]>([]);
     const [themeEdit, setThemeEdit] = useState<{ type: "edit" | "new"; theme?: Theme }>();
+
+    const [currentUser, setCurrentUser] = useState<User>();
+    getCurrentUser(api).then(user => setCurrentUser(user));
```

Comment:
Clean Arch: You should call a use case that returns the data you need to render in the component. Here, it does not seem you need `currentUser`, that's an intermediate value necessary to filter the rows. So the use case may look something like: `GetFilteredThemesUseCase` with `execute(themes: Theme[]): Theme[]`.

Reply:
i don't understand this comment much. would `GetFilteredThemesUseCase` return the current user as well? @xurxodev

Reply:
@deeonwuli Sorry, I don't understand the context of this code very well. so, I'm going to try to explain the goal of the use cases in clean architecture and I hope this help.
 
In Clean Architecture the use cases should work to execute actions or retrieve information. In the end, it's the real use cases of an app.

To retrieve information normally the data are entities of the domain, however, it's possible to retrieve related info together.

For example:
Working with an Invoice domain entity, you can have as use cases:
* GetInvoices
* GetInvoiceById
* SaveInvoice
* ...

These use cases use InvoiceRepository which return an Invoice domain entity or a list of invoice domain entities and request data against /invoices endpoint of an API.

But if you need to show a sales report for example you need to request against more endpoints, in this case, we can work with a new DTO SalesReport, GetSalesReport use case and SaleReportRepository that execute requests against different endpoints like: /invoices /clients /countries because need return more info than an invoice to can realize groupings.

Other times, you need to apply application logic to execute a use case.
For example, if an invoice cannot be created if the client is from France, the SaveInvoiceUseCase need to request data to two repositories: InvoiceRepository and ClientRepository, first need to retrieve the client to review the country and if it's not from France then execute InvoiceRepository.save (invoice). 

In the case of your code, I think that @arnau says that GetFilteredThemesUseCase returns themes and the themes are filtered according to the user groups but this filter is not realized by the component but by the use case.  In this case, the use case request also data to two repositories.

If you need to retrieve themes in several parts of the app, with the GetFilteredThemesUseCase avoid to duplicate the code to filter it.

I'm also not 100% sure because I don't understand the context very well of the code because I have never worked on this project.